src/CodeModel.cpp

```diff
@@ -212,72 +235,125 @@ void CodeModel::stop()
+		emit stateChanged();
+	}
+	catch (...)
+	{
+		manageException();
+	}
 }
 
 void CodeModel::unregisterContractSrc(QString const& _documentId)
 {
+	try
 	{
-		Guard pl(x_pendingContracts);
-		m_pendingContracts.erase(_documentId);
-	}
+		{
+			Guard pl(x_pendingContracts);
+			m_pendingContracts.erase(_documentId);
+		}
 
-	// launch the background thread
-	m_compiling = true;
-	emit stateChanged();
-	emit scheduleCompilationJob(++m_backgroundJobId);
+		// launch the background thread
+		m_compiling = true;
+		emit stateChanged();
+		emit scheduleCompilationJob(++m_backgroundJobId);
+	}
+	catch (...)
+	{
+		manageException();
+	}
 }
 
 void CodeModel::registerCodeChange(QString const& _documentId, QString const& _code)
 {
+	try
 	{
-		Guard pl(x_pendingContracts);
-		m_pendingContracts[_documentId] = _code;
-	}
+		{
+			Guard pl(x_pendingContracts);
+			m_pendingContracts[_documentId] = _code;
+		}
 
-	// launch the background thread
-	m_compiling = true;
-	emit stateChanged();
-	emit scheduleCompilationJob(++m_backgroundJobId);
+		// launch the background thread
+		m_compiling = true;
+		emit stateChanged();
+		emit scheduleCompilationJob(++m_backgroundJobId);
+	}
+	catch (...)
+	{
+		manageException();
+	}
 }
 
 QVariantMap CodeModel::contracts() const
 {
-	QVariantMap result;
-	Guard l(x_contractMap);
-	for (ContractMap::const_iterator c = m_contractMap.cbegin(); c != m_contractMap.cend(); ++c)
-		result.insert(c.key(), QVariant::fromValue(c.value()));
-	return result;
+	try
+	{
+		QVariantMap result;
+		Guard l(x_contractMap);
+		for (ContractMap::const_iterator c = m_contractMap.cbegin(); c != m_contractMap.cend(); ++c)
+			result.insert(c.key(), QVariant::fromValue(c.value()));
+		return result;
+	}
+	catch (...)
+	{
+		manageException();
+		return QVariantMap();
+	}
 }
 
 CompiledContract* CodeModel::contractByDocumentId(QString const& _documentId) const
 {
-	Guard l(x_contractMap);
-	for (ContractMap::const_iterator c = m_contractMap.cbegin(); c != m_contractMap.cend(); ++c)
-		if (c.value()->m_documentId == _documentId)
-			return c.value();
-	return nullptr;
+	try
+	{
+		Guard l(x_contractMap);
+		for (ContractMap::const_iterator c = m_contractMap.cbegin(); c != m_contractMap.cend(); ++c)
+			if (c.value()->m_documentId == _documentId)
+				return c.value();
+		return nullptr;
+	}
+	catch (...)
+	{
+		manageException();
+		return nullptr;
+	}
 }
 
 CompiledContract const& CodeModel::contract(QString const& _name) const
 {
-	Guard l(x_contractMap);
-	CompiledContract* res = m_contractMap.value(_name);
-	if (res == nullptr)
-		BOOST_THROW_EXCEPTION(dev::Exception() << dev::errinfo_comment("Contract not found: " + _name.toStdString()));
+	CompiledContract* res;
+	try
+	{
+		Guard l(x_contractMap);
+		res = m_contractMap.value(_name);
+		if (res == nullptr)
+			BOOST_THROW_EXCEPTION(dev::Exception() << dev::errinfo_comment("Contract not found: " + _name.toStdString()));
+	}
+	catch (...)
+	{
+		manageException();
+	}
 	return *res;
```

Comment:
This function was not directly called from JavaScript before, but could be.
This code is old, I think we can keep only one of them. @arkpar ?